### ISSUE TITLE
Unable to Disable "Thinking" When Fine-tuning Qwen3

### ISSUE THREAD

**Issue by kaizhuanren** [modelscope/ms-swift#4276]
I’m trying to fine-tune Qwen3 and would like to disable the “thinking” behavior. According to the documentation, I try

--model_kwargs '{"enable_thinking": false}'

However, this flag doesn’t seem to have any effect — the model still produces <thinking> during generation.

I suspect this might be because the enable_thinking argument is actually used in the tokenizer rather than the model itself. If that is the case, it is unclear how to correctly pass this argument when using the training framework 

So my questions are:

how should I stop thinking of model when fine-tuning the model?

To Reproduce:
# 2*24G GPU RTX4090
CUDA_VISIBLE_DEVICES=0,1 \
NPROC_PER_NODE=2 \
swift rlhf \
    --rlhf_type grpo \
    --model "Qwen3-8B" \
    --model_kwargs '{"enable_thinking": false}' \
    --train_type lora \
    --dataset "train.json" "val.json"\
    --torch_dtype bfloat16 \
    --num_train_epochs 10 \
    --max_length 8196 \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 1 \
    --eval_steps 100 \
    --save_steps 200 \
    --learning_rate 5e-5 \
    --save_total_limit 500 \
    --logging_steps 1 \
    --output_dir output_aws \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 2 \
    --max_completion_length 8192 \
    --reward_funcs debug_completion overall_length format_strict format_soft acc \
    --num_generations 4 \
    --use_vllm true \
    --vllm_gpu_memory_utilization 0.6 \
    --vllm_max_model_len 2048 \
    --deepspeed zero3_offload \
    --temperature 1.0 \
    --top_p 1.0 \
    --top_k 80 \
    --log_completions true \
    --num_infer_workers 2 \
    --tensor_parallel_size 2 \
    --async_generate false \
    --move_model_batches 16 \
    --offload_optimizer true \
    --offload_model true \
    --gc_collect_after_offload true \
    --sleep_level 0 \
    --seed 42 \
    --external_plugins pl/plugin.py

**Comment by Jintao-Huang** [comment]
`--response_prefix '<think>\n\n</think>\n\n'`

**Comment by Jintao-Huang** [comment]
--template qwen3_nothinking
